pkg/build/gcloud/storage/gsutil.go

```diff
@@ -366,11 +366,17 @@ func GetLatestMainBuild(ctx context.Context, bucket *storage.BucketHandle, path
 		return "", ErrorNilBucket
 	}
 
-	it := bucket.Objects(ctx, &storage.Query{
+	query := &storage.Query{
 		Prefix: path,
-	})
+	}
+	err := query.SetAttrSelection([]string{"Name", "Generation"})
```

Comment:
I think it would be better to use the `Updated` attribute here, then do filtering of the name in the first loop and fill everything into a slice to sort. This way IMO the code starting at line 388 would become much more readable 🙂

Reply:
Hm, why `Updated` is better than `Generation`? 🤔 
Grafana uploaded packages should be immutable I think? 

Also, about 
> filtering of the name in the first loop and fill everything into a slice to sort

I am doing the sort while I am constructing the slice so we won't have to go and run a sort on a slice afterwards. It's more performant IMO and that's why I wanted to go with it this way. I can always add some comments if it makes it easier for people to read and understand it.
Feel free to object though 😄

Reply:
`Updated` for unchanged items is the same as `Created` :)

But yes, manual sorting is also fine for me :)